game/songitems.cc

```diff
@@ -71,17 +71,19 @@ void SongItems::addSong(std::shared_ptr<Song> song) {
 	m_songs.insert(si);
 }
 
-int SongItems::lookup(std::shared_ptr<Song> song) const {
-	for (auto const& s: m_songs) {
-		if (song->collateByArtistOnly == s.artist && song->collateByTitleOnly == s.title) return s.id;
-	}
+int SongItems::lookup(std::shared_ptr<Song> const& song) const {
+	for (auto const& s: m_songs)
+		if (song->collateByArtistOnly == s.artist && song->collateByTitleOnly == s.title) 
+			return s.id;
+
 	return -1;
```

Comment:
this id seems to eventually be used as a unsigned; maybe return an std::optional<> ?

Reply:
I pushed the change. I am not sure if it is a good solution with std::optional. @OznOg please take a look.